messaging/kafka/client/syncproducer.go

```diff
@@ -37,8 +37,9 @@ type SyncProducer struct {
 	sync.Mutex
 }
 
-// NewSyncProducer returns a new SyncProducer
-func NewSyncProducer(config *Config, wg *sync.WaitGroup) (*SyncProducer, error) {
+// NewSyncProducer returns a new SyncProducer instance. Producer is created from provided sarama client. Also
+// the partitioner is set here. Note: sarama partitioner should match the one used in config.
+func NewSyncProducer(config *Config, sClient sarama.Client, partitioner string, wg *sync.WaitGroup) (*SyncProducer, error) {
```

Comment:
Please, either handle in this function case when sClient is nil and create own new client that will be used or rename this one to `NewSyncProducerFromClient` and handle the case in separate function.

See also the comment about closing the client.